lib/resultuploaders/result_uploader.rb

```diff
@@ -10,24 +11,38 @@ class ResultUploader
     # UploaderFactory
     #
     class UploaderFactory
-      UPLOADERS = {
-        dropbox: Dropbox
-      }.freeze
+      @uploaders = {}
```

Comment:
Now I see no reason for this conversion from constant to variable.

Reply:
Sorry, what do you mean? 
Drop `@uploaders` and rewrite the `create` function? 

```ruby
      def self.create(type, project)
        AutoHCK.const_get(type.camelize).new(project)
      end
 ```

Reply:
You can write:

```suggestion
      UPLOADERS = {}
```

Reply:
Yep

Reply:
In this case of the UPLOADERS constant, we got the RuboCop offense `Freeze mutable objects assigned to constants.Style/MutableConstant(RuboCop)`. We can't freeze this hash because it will be updated.

Reply:
You can (and perhaps you should) freeze it after the hash is filled.

Reply:
But RuboCop asked to freeze during objects assigned.

Reply:
No, because we have return for files. to_h fail in this case, or you suggest to return [nil,nil] in this case?

Reply:
You can first have an `Enumerator` for paths under `__dir__`, and then call `select` to select directories.

Reply:
Fixed